pkgs/build-support/cc-wrapper/default.nix

```diff
@@ -154,6 +153,7 @@ stdenv.mkDerivation {
         local dst="$1"
         local wrapper="$2"
         export prog="$3"
+        export isdarwin="${toString targetPlatform.isDarwin or false}"
```

Comment:
as I almost expected, this does not work, the substitution is not performed in the wrapper script.